app/scripts/modules/core/src/presentation/forms/stories/FormikIntro.stories.mdx

```diff
@@ -0,0 +1,117 @@
+
+import { FormikFormField } from '../';
+import {URLComponent} from './SpinFormikStoriesHelper'
+
+<Meta
+  title="Forms/Intro"
+  decorators={[
+    (Story) => (
+      <div style={{ background: '#fff', border: '1px solid #eee', borderRadius: 10, padding: 20, width: 500 }}>
+        <Story />
+      </div>
+    ),
+  ]}
+  parameters={{
+    layout: 'centered',
+  }}
+/>
+
+Spinnaker forms are built upon [Formik](https://formik.org) and provides two key abstractions for easily building forms: `SpinFormik` and `FormikFormField`.
+
+## `SpinFormik`
+
+This component is a thin wrapper around [`Formik`](https://formik.org/docs/api/formik) component. A typical form is created the following way
+
+```jsx
+interface FormValues {
+  field1: string;
+  field2: number;
+}
+
+<SpinFormik<FormValues>
+  initialValues={{ field1: 'foo', field2: 2 }}
+  render={(formikProps) => <YourFormView {...formikProps} />}
+  onSubmit={(values) => {
+    /* handle form submission */
+  }}
+/>
+```
+
+### Props
+
+`SpinFormik` passes all props to `Formik`, so check out the [documentation](https://formik.org/docs/api/formik#reference) for detailed information on props accepted by `Formik`.
+
+## `FormikFormField`
+
+This component handles the communication with formik and provides the necessary props to your input fields. Render your fields by passing them to the `input` prop.
+
+```jsx
+<SpinFormik<FormValues>
+  initialValues={{ field1: 'foo', field2: 2 }}
+  render={(formikProps) => (
+    <form onSubmit={formikProps.handleSubmit}>
+      <FormikFormField name="field" label="Field 1" input={(inputProps) => <TextInput {...inputProps} />} />
+    </form>
+  )}
+  onSubmit={(values) => {
+    /* handle form submission */
+  }}
+/>
+```
+
+### Props
+
+<ArgsTable of={FormikFormField} />
+
+Here's a simple form with all supported input fields
+
+<!-- Simple Form Example -->
+<Canvas>
+  <Story id="forms-spinformik--simple-form" />
+</Canvas>
+
+## Validation
+
+Forms can be validated by providing a `validate` function to `FormikFormField`. The validate function is expected to return an error message when the field value is invalid and `FormikFormField` automatically renders the error message.
+If you wish to use the validation errors outside the scope of a field, then `formikProps` (argument provided via the `render` function) provides `errors` as a map of all the errors keyed by their field name.
+In the following example, the errors object is used to disable the submit button. More details on formik validation can be found [here](https://formik.org/docs/guides/validation).
+
+
+<!-- Validation Example -->
+<Canvas>
+  <Story id="forms-spinformik--validation" />
+</Canvas>
+
+## Custom Field Input
+
+Spinnaker provides standard form input fields that cover most common scenarios. If your use case is unique, then you can create a custom input field and use that in the `input` prop to `FormikFormField`.
```

Comment:
I think this also should be less prominently featured somehow.

Reply:
I'm going to handle it in a separate PR. I agree it should be less prominent, but I also want to avoid removing it completely from documentation as there is some value in showing how an existing field can be modified to work with `FormikFormField`.